mlir/include/mlir/Dialect/Transform/IR/TransformOps.td

```diff
@@ -459,6 +459,36 @@ def NumAssociationsOp : TransformDialectOp<"num_associations",
   }];
 }
 
+def CollectMatchingOp : TransformDialectOp<"collect_matching", [
+    DeclareOpInterfaceMethods<MemoryEffectsOpInterface>,
+    DeclareOpInterfaceMethods<SymbolUserOpInterface>,
+    DeclareOpInterfaceMethods<TransformOpInterface>]> {
+  let summary = "Collects all payload ops that match the given named matcher";
+  let description = [{
+    Collects operations nested under `root` or other payload IR objects that
```

Comment:
I would mention that `root` itself is not taken into account when matching ops.

Reply:
It is taken into account, the implementaiton does a plain `walk`. Do we want an option that doesn't include root?

Reply:
`-> Collects operations nested under `root` (included)` ?